lib/pipelines/code-pipeline.ts

```diff
@@ -32,6 +33,26 @@ export interface GitHubSourceRepository extends Omit<ApplicationRepository, "cre
     credentialsSecretName: string
 }
 
+/**
+ * Props for the CodeCommit repository.
+ */
+export type CodeCommitProps = {
```

Comment:
For consistency with the GitHub repo let's create an interface `CodeCommitSourceRepository` with properties `codeCommitRepoName`. 

This can enable us to declare the `repository` field in the PipelineProps as:

```
repository: GitHubSourceRepository | CodeCommitSourceRepository
```

This field will be required and remove the need to check when both are specified.